unicore/hub/service/utils.py

```diff
@@ -1,8 +1,63 @@
+import re
 import os
 import base64
 
+from zope.interface import implementer
+from pyramid.interfaces import IAuthenticationPolicy
+
 
 def make_password(bit_length=15):
     """ Returns a random password string
     """
     return base64.b64encode(os.urandom(bit_length))
+
+
+@implementer(IAuthenticationPolicy)
+class PathAuthenticationPolicy(object):
```

Comment:
It feels like it needs to be in a file called `policy.py` or `authentication.py` instead of utils.